plugin/chatgpt.vim

```diff
@@ -48,6 +26,59 @@ endif
 if !exists("g:chat_gpt_split_direction")
   let g:chat_gpt_split_direction = 'horizontal'
 endif
+if !exists("g:split_ratio")
+  let g:split_ratio = 3
+endif
+
+" Add ChatGPT dependencies
+python3 << EOF
+import sys
+import vim
+import os
+
+try:
+    from openai import AzureOpenAI, OpenAI
+except ImportError:
+    print("Error: openai module not found. Please install with Pip and ensure equality of the versions given by :!python3 -V, and :python3 import sys; print(sys.version)")
+    raise
+
+def safe_vim_eval(expression):
+    try:
+        return vim.eval(expression)
+    except vim.error:
+        return None
+
+def create_client():
+    api_type = safe_vim_eval('g:api_type')
+    api_key = os.getenv('OPENAI_API_KEY') or safe_vim_eval('g:chat_gpt_key') or safe_vim_eval('g:openai_api_key')
+
+    openai_base_url = safe_vim_eval('g:openai_base_url')
+    azure_endpoint = safe_vim_eval('g:azure_endpoint')
+    azure_api_version = safe_vim_eval('g:azure_api_version')
+    azure_deployment = safe_vim_eval('g:azure_deployment')
+    if api_type == 'azure':
+        assert azure_endpoint and azure_api_version and azure_deployment, "plz set azure_endpint, azure_api_version and azure_deployment"
+        assert api_key, "plz conif api_key"
+        client = AzureOpenAI(
+            azure_endpoint=azure_endpoint,
+            azure_deployment=azure_deployment,
+            api_key=api_key,
+            api_version=azure_api_version,
+        )
+    elif api_type == 'openai':
```

Comment:
This should be the default, so the additional config doesn't break the setup for people who are not adopting the AzureOpenAI.  At present this will result in `client` not being defined for existing users, until they set `api_type`.